lib/usbmux/index.js

```diff
@@ -171,18 +180,43 @@ class Usbmux extends BaseServiceSocket {
     if (!device) {
       throw new Error(`Could not find the expected device '${udid}'`);
     }
-    const plistService = new PlistService(await this.connect(device.Properties.DeviceID, LOCKDOWN_PORT, timeout));
-    return new Lockdown(plistService);
+    return await this.connect(device.Properties.DeviceID, LOCKDOWN_PORT, timeout, (socket) => new Lockdown(new PlistService(socket)));
   }
 
+  /** The callback function which will be when connect succeeds synchronously
+   * @name ConnectCallback
+   * @function
+   * @param {net.Socket} socket the socket for the communication to the phone
+   * @returns {Object} The object returned in the callback function
+  */
+
   /**
    * Connects to a certain port on the device
    * @param {string} deviceID the device id which can be retrieved from the properties of a device
    * @param {number} port the port number that wants to be connected
    * @param {number} [timeout=5000] the timeout of receiving a response from usbmuxd
+   * @param {?ConnectCallback} callback The callback function which will be when connect succeeds synchronously
+   * @returns {net.Socket|Object} The socket or the object returned in the callback if the callback function exists
    */
-  async connect (deviceID, port, timeout = 5000) {
-    const { tag, receivePromise } = this._receivePlistPromise(timeout);
+  async connect (deviceID, port, timeout = 5000, callback) {
+    const {tag, receivePromise} = this._receivePlistPromise(timeout, (data) => {
+      if (data.payload.MessageType !== 'Result') {
+        throw new Error(`Unexpected data: ${JSON.stringify(data)}`);
+      }
+      if (data.payload.Number === USBMUX_RESULT.OK) {
+        this._splitter.shutdown();
+        this._socketClient.unpipe(this._splitter);
+        this._splitter.unpipe(this._decoder);
+        if (!callback) {
```

Comment:
return _.isFunction(callback) ? callback(this._socketClient) : this._socketClient